src/webapi/date.rs

```diff
@@ -1,22 +1,903 @@
+        js!( return @{self.as_ref()}.getUTCDay(); ).try_into().unwrap()
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCFullYear)
+    #[inline]
+    fn get_utc_full_year( &self ) -> f64 {
+        js!( return @{self.as_ref()}.getUTCFullYear(); ).try_into().unwrap()
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCHours)
+    #[inline]
+    fn get_utc_hours( &self ) -> f64 {
+        js!( return @{self.as_ref()}.getUTCHours(); ).try_into().unwrap()
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCMilliseconds)
+    #[inline]
+    fn get_utc_milliseconds( &self ) -> f64 {
+        js!( return @{self.as_ref()}.getUTCMilliseconds(); ).try_into().unwrap()
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCMinutes)
+    #[inline]
+    fn get_utc_minutes( &self ) -> f64 {
+        js!( return @{self.as_ref()}.getUTCMinutes(); ).try_into().unwrap()
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCMonth)
+    #[inline]
+    fn get_utc_month( &self ) -> f64 {
+        js!( return @{self.as_ref()}.getUTCMonth(); ).try_into().unwrap()
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCSeconds)
+    #[inline]
+    fn get_utc_seconds( &self ) -> f64 {
+        js!( return @{self.as_ref()}.getUTCSeconds(); ).try_into().unwrap()
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getYear)
+    #[inline]
+    fn get_year( &self ) -> f64 {
+        js!( return @{self.as_ref()}.getYear(); ).try_into().unwrap()
+    }
+
+    /// NOTE: for the following functions that take arguments in addition to self, the arguments
+    /// are allowed to be empty. From the user's perspective, the first argument is mandatory, with
+    /// the rest optional. However, browsers do not need all arguments, even the first, but calling
+    /// with empty arguments returns NaN.
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setDate)
+    #[inline]
+    fn set_date( &self, _day_value: f64 ) -> f64 {
+        js!( return @{self.as_ref()}.setDate(@{_day_value}); ).try_into().unwrap()
+    }
+
+    ///
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setFullYear)
+    #[inline]
+    fn set_full_year( &self, args: &[i32] ) -> f64 {
```

Comment:
It would be better to just take `year: i32, month: Option< i32 >, day: Option< i32 >` here.